### PR TITLE
Fix wrong comparison with ipV6

### PR DESCRIPTION
**Pull request by minsulee2**
I made the wrong design for ip address comparison.

- Remove wrong converting
- Compare ip addresses correctly

### CODE REVIEW

File: p2p/server.js [hunk 1]
```diff
@@ -391,11 +391,12 @@ class P2pServer {
   /**
    * Returns true if the socket ip address is the same as the given p2p url ip address,
    * false otherwise.
-   * @param {string} ipv4Address is ipv4 socket._socket.remoteAddress
-   * @param {string} url is peerInfo.networkStatus.urls.p2p.url
+   * @param {string} ipAddress can be both ipv4 or ipv6 socket._socket.remoteAddress.
```
Comment (platfowner) [inline p2p/server.js:394]: minor) s/both/either?

File: p2p/server.js [hunk 2]
```diff
@@ -391,11 +391,12 @@ class P2pServer {
...
+   * @param {string} url is peerInfo.networkStatus.urls.p2p.url.
    */
-  checkIpAddressFromPeerInfo(ipv4Address, url) {
-    return url.includes(ipv4Address);
+  checkIpAddressFromPeerInfo(ipAddress, url) {
```
Comment (platfowner) [inline p2p/server.js:397]: what about defining it static in p2p-util.js and add some test cases to p2p-util.test.js?

File: p2p/server.js (hunk 1 above)
Comment (minsulee2) [reply p2p/server.js:394]: Oh, fixed!

File: p2p/server.js [hunk 3]
```diff
@@ -391,11 +391,12 @@ class P2pServer {
...
+    const fromUrl = new URL(url);
```
[the remark this answers is not in the document]
Comment (minsulee2) [reply p2p/server.js]: Done!

File: p2p/server.js (hunk 2 above)
Comment (minsulee2) [reply p2p/server.js]: Gooooood idea! job done!

**Comment by minsulee2** [comment]
PTAL All addressed

File: db/rule-util.js [hunk 4]
```diff
@@ -399,6 +399,11 @@ class RuleUtil {
     return this.isString(url) ? privateUrlRegex.test(url) : false;
   }
 
+  isValidIpV6(ipAddress) {
```
Comment (platfowner) [inline db/rule-util.js:402]: could you add some test cases to rule-util.test.js?

File: p2p/p2p-util.js [hunk 5]
```diff
@@ -156,6 +157,35 @@ class P2pUtil {
...
+    try {
+      const fromUrl = new URL(url);
+      return fromUrl.hostname;
+    } catch (e) {
+      return null;
+    }
+  }
+
+  static isValidIpAddress(ipAddress) {
+    return CommonUtil.isValidUrl(ipAddress) || CommonUtil.isValidPrivateUrl(ipAddress) ||
```
Comment (platfowner) [inline p2p/p2p-util.js:171]: are urls (e.g. https://172.20.10.2) allowed? if not, could you add and use isValidIpV4() instead?

File: test/unit/p2p-util.test.js [hunk 6]
```diff
@@ -302,4 +302,85 @@ describe("P2P Util", () => {
...
+      const validHttpValue = 'http://172.20.10.2:8080';
+      const validWsValue = 'ws://172.20.10.2:8080';
+
+      expect(util.toHostname(validHttpValue)).to.equal('172.20.10.2');
+      expect(util.toHostname(validWsValue)).to.equal('172.20.10.2');
+    });
+  });
+
+  describe("isValidIpAddress", () => {
+    it("returns false if invalid ipAddress is given", () => {
```
Comment (platfowner) [inline test/unit/p2p-util.test.js:337]: could you add full url test cases (e.g. https://172.20.10.2, ws://172.20.10.2, https://172.20.10.2:8080, ...) somewhere?

File: db/rule-util.js (hunk 4 above)
Comment (minsulee2) [reply db/rule-util.js:402]: Done !

File: p2p/p2p-util.js (hunk 5 above)
Comment (minsulee2) [reply p2p/p2p-util.js]: Oh, nice point! Updated and added isValidIpV4 as well !

File: test/unit/p2p-util.test.js (hunk 6 above)
Comment (minsulee2) [reply test/unit/p2p-util.test.js:337]: Updated!

**Review by platfowner — APPROVED** [review]
LGTM
Thanks for the nice refactoring!

**Comment by minsulee2** [comment]
Thanks for the review! Merging now :)

**Pull request merged**